src/lib/Microsoft.Health.Events/Telemetry/Exceptions/EventHubExceptionTelemetryProcessor.cs

```diff
@@ -14,35 +14,81 @@ namespace Microsoft.Health.Events.Telemetry.Exceptions
 {
     public static class EventHubExceptionTelemetryProcessor
     {
+        private static readonly EventHubConfigurationExceptionTelemetryProcessor _exceptionTelemetryProcessor = new EventHubConfigurationExceptionTelemetryProcessor();
+
         public static void ProcessException(
             Exception exception,
             ITelemetryLogger logger,
-            bool shouldLogMetric = true,
             string errorMetricName = null)
         {
             EnsureArg.IsNotNull(exception, nameof(exception));
             EnsureArg.IsNotNull(logger, nameof(logger));
 
-            logger.LogError(exception);
+            var ex = CustomizeException(exception);
+
+            logger.LogError(ex);
 
-            if (shouldLogMetric)
+            if (ex.Equals(exception))
             {
-                logger.LogMetric(EventMetrics.HandledException(errorMetricName ?? GetErrorMetricName(exception), ConnectorOperation.Setup), 1);
+                logger.LogMetric(
+                    EventMetrics.HandledException(
+                        errorMetricName ?? $"{ErrorType.EventHubError}{EventHubErrorCode.GeneralError}",
+                        ConnectorOperation.Setup),
+                    1);
+            }
+            else
+            {
+                _exceptionTelemetryProcessor.HandleException(ex, logger, ConnectorOperation.Setup);
             }
         }
 
-        private static string GetErrorMetricName(Exception exception)
+        public static Exception CustomizeException(Exception exception)
         {
+            string message = exception.Message;
+
             switch (exception)
             {
                 case EventHubsException _:
-                    return $"{ErrorType.EventHubError}{EventHubErrorCode.OperationError}";
+                    EventHubsException eventHubsException = (EventHubsException)exception;
+                    switch (eventHubsException.Reason)
+                    {
+                        case EventHubsException.FailureReason.ConsumerDisconnected:
+                            message = "Verify that the provided Event Hub's consumer group is not already receiving data for another IoT connector or Azure resource.";
+                            break;
+                        case EventHubsException.FailureReason.ResourceNotFound:
+                            message = "Verify that the provided Event Hubs Namespace contains the provided Event Hub and that the provided Event Hub contains the provided consumer group.";
+                            break;
+                        case EventHubsException.FailureReason.ServiceCommunicationProblem:
+                            message = "Verify that the provided Event Hub Namespace, Event Hub name, and consumer group are correct and that access permissions to the provided Event Hub have been granted.";
+                            break;
+                        default:
+                            return exception;
```

Comment:
Sorry didn't ask this in 1st iteration - for these exceptions that we are processing, if we should use the FailureReason/SocketErrorCode in the default case too. 
I think for exceptions that we are not processing having a metric `EventHubErrorGeneralError` is fine, since we don't know what the exception type is, so in the default case at line 91 this metric is fine.
However for the defaults within each of these exceptions it would be nice to still have the FailureReason/SocketErrorCode captured in the metric (instead of just GeneralError) too so we can get visibility into what the various types of errors we run into.

Reply:
Good idea. I initially thought the inner exception would tell us the reason/socketError but having it shown in the Metric Name would be helpful in identifying, at a glance, errors we haven't come across yet.
Updated code accordingly.